### PR TITLE
Split "Quick Start" docs into their own section.

### PR DESCRIPTION
**Pull request by Tahler**
Should address some confusion and part of #1299.

https://docs.google.com/document/d/1ks8lpbK1lOh1HKLqcvo9QOcyeFwWcx5o7MB6A9-ycr0/ for more info.

### CODE REVIEW

**Comment by geeknoid** [comment]
Could you provide a staging site so we can look at the rendered content? You can do this by pointing to your GItHub fork via GitHub pages.

Thanks.

**Comment by Tahler** [comment]
[There](https://tahler.github.io/istio.github.io/docs/). Had to merge it to my master branch and took a bit to actually publish.

**Comment by linsun** [comment]
Agreed, I'd appreciate a staging site :)  I think it is a good idea to move up Quick Start up to one level @geeknoid

**Comment by Tahler** [comment]
@linsun You can see the staging site here:
image

@geeknoid Thanks for the feedback. I definitely see what you're talking about. I still think there's a problem with the current layout, however. The setup section is rather disorganized and several people were frustrated with that. Even in the description, 

    Setup contains instructions for installing the Istio control plane in various 
    environments, as well as instructions for installing the sidecar in the application
    deployment. Quick start instructions are available for Kubernetes and Docker 
    Compose w/ Consul.

it seems to me the setup section has too much going on. It mixes quick start / experimental installation instructions with production installation instructions. The sidecar and mesh expansion docs depend on other docs.

The solution I came up was to make the layout more hierarchical and decision-tree-like. I'm curious to hear your thoughts on that tree in the [doc](https://docs.google.com/document/d/1ks8lpbK1lOh1HKLqcvo9QOcyeFwWcx5o7MB6A9-ycr0/edit).

At the very least, I think there should be more clear focus in the docs what user is going to be reading them (e.g. Is this doc only for users who are wanting to experiment? Is this doc only for people who already have Istio installed on their cluster?) and it should be quick for a user to find the installation doc for their use case. In its current state, a new user basically has to linear search through the documents or restart their progress when they realize they didn't want to install a certain way.

tl;dr I think the setup needs more hierarchy and focus.

**Comment by linsun** [comment]
Thanks, preview was neat.

Should we at least start with Kubernetes as the first item in quick start?  Also, it is probably good to have each provider have its own quick start so the quick start could be short and focus on docker for k8s or minikube (if works).

**Comment by andraxylia** [comment]
@Tahler where is the staging site? I do not see a link above.

**Comment by andraxylia** [comment]
Staging site is here: https://deploy-preview-1338--preliminary-istio.netlify.com/

**Comment by linsun** [comment]
I think this layout is good.  My only comment is can we have Kubernetes as the first item in quick start?

**Comment by linsun** [comment]
also, pls resolve merge conflict so we could get this in :)

**Comment by Tahler** [comment]
Are we sure we want to get this in? I don't think we resolved @geeknoid's thoughts on it so we don't have consensus on this.

**Comment by geeknoid** [comment]
Since the site has changed so much, I'm closing this PR. Please resubmit with updates if you think this is valuable.

Thanks.
